alibi_detect/version.py

```diff
@@ -2,4 +2,6 @@
 # 1) we don't load dependencies by storing it in __init__.py
 # 2) we can import it in setup.py for the same reason
 # 3) we can import it into your module module
-__version__ = "0.9.2dev"
+# The version is read from the pyproject.toml file
+from importlib import metadata
+__version__ = metadata.version(__package__)
```

Comment:
How does this actually work? Is `importlib.metadata` a representation of `pyproject.toml` for the package? How does `pyproject.toml` get packaged into the package during build (e.g. no need to include in `MANIFEST.in`? Does some magic happen during build so that `pyproject.toml` is mapped to `importlib.metadata`?

Reply:
This is following the example here: https://github.com/pypa/setuptools_scm#retrieving-package-version-at-runtime

It relies on the package actually being installed (and yes `pyproject.toml` does seem to packaged by default during the build). At the very least I should probably add a `PackageNotFoundError` like in the example... 

I've actually just pushed an update using `setuptools_scm`. This approach has two main benefits IMO:

1. It would simplify the release workflow slightly. We'd just create the new git tag, and then do the build. No need to update hardcoded version number (but still need to update readme annoyingly). 
2. There would be no possibility of accidentally publishing dirty code. If you install/build with changes in the `workdir`, the version will be set to something like `0.9.2.dev16+g68e502c.d20220509` (`g` stands for git commit, `d` for date). PyPI doesn't accept these "dirty" version numbers. 
3. Above "dirty" version numbers would also be written to generate `config.toml` files. This would be helpful as it would indicate whether or not a detector was generated from a non-official version of Alibi Detect.

Reply:
Note: `alibi_detect._version.py` is removed from github tracking in order to use `setuptools_scm`. However, this will require `alibi_detect` to be installed on readthedocs #499 (see [setuptools_scm discussion](https://github.com/pypa/setuptools_scm/#usage-from-sphinx)). This might make sense to do anyway after the optional deps project...